src/main/java/com/splunk/logging/serialization/HecJsonSerializer.java

```diff
@@ -63,6 +63,10 @@ public String serialize(HttpEventCollectorEventInfo info) {
         } else {
             event.put("event", info);
         }
+        if (!event.containsKey("time") && info.getTime() > 0) {
```

Comment:
Maybe it is worth to create a separate method for that part one the time is not specified?